packages/block-library/src/list-item/hooks/use-merge.js

```diff
@@ -107,11 +107,14 @@ export default function useMerge( clientId, onMerge ) {
 			} else if ( previousBlockClientId ) {
 				const trailingId = getTrailingId( previousBlockClientId );
 				registry.batch( () => {
-					moveBlocksToPosition(
-						getBlockOrder( clientId ),
-						clientId,
-						previousBlockClientId
-					);
+					const [ nestedListClientId ] = getBlockOrder( clientId );
```

Comment:
Can we add a quick comment to refer to in the future? It's not immediately clear from the logic alone why we are doing this

Reply:
Added a comment